python/sparkdl/__init__.py

```diff
@@ -13,8 +13,18 @@
 # limitations under the License.
 #
 
+
+# hack to import copy-pasted image schema (to be removed in Spark2.3)
+# TODO remove in Spark2.3
+import os
+import pyspark.ml
+dir_path = os.path.dirname(os.path.realpath(__file__))
+parentdir = os.path.dirname(dir_path)
+pyspark.ml.__path__.append(os.path.join(parentdir, "pyspark", "ml"))
+
+from pyspark.ml.image import ImageSchema
+
```

Comment:
I'm concerned that monkey patching the spark.ml might introduce unexpected behaviour for the user. Specifically I'm worried that this will work?

```
import sparkdl
from spark.ml.image import ImageSchema
```

but this will not:
```
from spark.ml.image import ImageSchema
import sparkdl
```

The alternative is to put this in `sparkdl.something`, and we can update `sparkdl.something` to be an alias for `pyspark.ml.image` for newer versions of spark.

I don't have a preference here, I just want to make sure both options are considered.

Reply:
My take is we have two options here:
    1. do the monkey patch
        plus: no change in code needed after switching to spar2.3
        minus: user must import sparkdl first
   2. change the module of copy pasted files  to sparkdl .image
       plus: standard python imports, order of imports does not matter
       minus: need to change code after spark 2.3

I have slight preference for 1 since it has the nice property that your script will keep working once you get it work first time.

Reply:
If we simply add `from pyspark.ml import image` to sparkdl.__init__ for => spark2.3 then we'll get  the "just keep working" property with option two. We've done this in dipy when we move stuff around internally.

Reply:
Yeah ok, that works.I can make the change.

Reply:
Actually, I still find it confusing that you end up in a state that you can mix imports of sparkdl.image.ImageSchema and some pyspark.ml.image.ImageSchema and they are actually the same thing. It might be better than the monkey patch, however the monkey patch is only temporary while this solution would persist.